### PR TITLE
Preparations for the upgrade to Scala 2.13.0-M4.

### PR DESCRIPTION
**Pull request by sjrd**
This PR contains commits that get the codebase closer to a state where it can support 2.13.0-M4, by improving things that should have been done already anyway.

Most of it is extracted from #3344.

### CODE REVIEW

File: javalib/src/main/scala/java/math/BigDecimal.scala [hunk 1]
```diff
@@ -194,8 +194,15 @@ object BigDecimal {
...
-  private[math] def newArrayOfPows(len: Int, pow: Int): Array[Long] =
-    new Array[Long](len - 1).scanLeft[Long, Array[Long]](1)((z, e) => z * pow)
+  private[math] def newArrayOfPows(len: Int, pow: Int): Array[Long] = {
+    val result = new Array[Long](len)
+    var nextValue = 1L
+    for (i <- 0 until len) {
+      result(i) = nextValue
+      nextValue *= pow
+    }
+    result
```
Comment (gzm0) [inline javalib/src/main/scala/java/math/BigDecimal.scala]: Why not just `Array.iterate(1L, len)(_ * pow)`? Shouldn't this expand into the same anyways past optimizer?

File: test-interface/src/main/scala/org/scalajs/testinterface/TestDetector.scala [hunk 2]
```diff
@@ -53,7 +53,7 @@ private[scalajs] object TestDetector {
     def tryLoad(name: String): Option[Framework] =
       tryLoadFromReflect(name).orElse(tryLoadFromExportsNamespace(name))
 
-    names.toStream.map(tryLoad).flatten.headOption
+    names.iterator.map(tryLoad).find(_.isDefined).flatten
```
Comment (gzm0) [inline test-interface/src/main/scala/org/scalajs/testinterface/TestDetector.scala]: have you considered using `collectFirst { Some(f) => f }` for better readability?

**Comment by sjrd** [comment]
Added a commit with the removal of procedure syntax in examples.

**Comment by sjrd** [comment]
@gzm0 Anything else here?

**Pull request merged**